From: Simon McVittie <smcv@collabora.com>
Date: Tue, 10 Jan 2023 13:40:24 +0000
Subject: garray: Avoid calling memcpy with no items

`memcpy(NULL, ., n)` and `memcpy(., NULL, n)` are undefined behaviour,
even if *n* is zero.

When len is 0 here, callers are allowed to pass in null data, and
GPtrArray also does not guarantee to have allocated rarray->pdata yet.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3192
---
 glib/garray.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/garray.c b/glib/garray.c
index 334a592..39fc3c6 100644
--- a/glib/garray.c
+++ b/glib/garray.c
@@ -1382,7 +1382,7 @@ ptr_array_new_from_array (gpointer       *data,
       for (gsize i = 0; i < len; i++)
         rarray->pdata[i] = copy_func (data[i], copy_func_user_data);
     }
-  else
+  else if (len != 0)
     {
       memcpy (rarray->pdata, data, len * sizeof (gpointer));
     }
